spec.bs

```diff
@@ -310,21 +343,189 @@ maps to an internal [=fenced frame config=] [=struct=].
 Issue: Fix the "indistinguishable" IDL bug with the unions above.</span>
 
 Each {{FencedFrameConfig}} has a <dfn for=fencedframeconfig>url</dfn>, which is a string, initially
-null.
+null. TODO: flesh out {{FencedFrameConfig}}
+
+A key feature of the <{fencedframe}> element is that web platform APIs can configure the behavior
+of the frame in a way that limits the ability of other execution contexts to modify or inspect this
+configuration, for security or privacy reasons. For example, the FLEDGE API (TODO:ref) performs
+on-device ad auctions over cross-site data, and it is important that the ad that wins the auction
+can be loaded into a frame, without the API caller knowing which ad won the auction or being able
+to manipulate the environment in which the ad loads.
+
+We achieve this using the concept of a "fenced frame config". A fenced frame config is a collection
+of fields that can be loaded into <{fencedframe}> elements and that specifies the resulting
+environments. Fenced frame configs can only be constructed by web platform APIs, not initialized
+or modified arbitrarily. Their fields also contain "visibilities" for different entities, which
+dictate whether the field should be "redacted" when inspected by particular execution contexts.
+Each time a fenced frame config is loaded into a <{fencedframe}> element, it is instantiated as a
+new "fenced frame config instance", which governs that particular context.
+
+We now establish some preliminary types:
+
+A <dfn export for=fencedframeconfig>visibility</dfn> is either
```

Comment:
Will anyone ever directly reference this dfn instead of the two embedder- and content-specific ones? If not, maybe we can just define both types as having the generic values defined below.

Reply:
There may be places in the redaction spec where we want to phrase things generally in terms of visibilities. But outside of the FF spec, I don't think so.

Reply:
I think we actually don't need visibility to content, like I suspected earlier, but let me finish fleshing out all the fields/redaction and see if that's still the case.